### ISSUE TITLE
Can't overwrite default inventory hosts file in .debops.cfg

### ISSUE THREAD

**Issue by dvigueras** [debops/debops-tools#125]
When trying to overwrite the default `hostfile` var in `.debops.cfg`, after generating `ansible.cfg` the value is still the default value:

Contents of `.debops.cfg`:

```
[ansible defaults]
hostfile = ./provisioning/vagrant_ansible_inventory
```

Value of hostfile in generated `ansible.cfg`:

```
# Ansible configuration file generated by DebOps, all changes will be lost.
# You can manipulate the contents of this file via `.debops.cfg`.

[defaults]
(other vars)
hostfile = /Users/daniel/devel/dvigueras/vagrant-debops/ansible/inventory
(other vars)
```

I've found that if I delete line 84 in debops: https://github.com/debops/debops/blob/master/bin/debops#L84 the file gets generated correctly:

```
# Ansible configuration file generated by DebOps, all changes will be lost.
# You can manipulate the contents of this file via `.debops.cfg`.

[defaults]
(other vars)
hostfile = ./provisioning/vagrant_ansible_inventory
(other vars)
```

And even ansible works, but debops isn't able to connect to the vm:

``` bash
$ ansible -m ping all
vagrantdebops | success >> {
    "changed": false,
    "ping": "pong"
}

$ debops-task -m ping all
No hosts matched
```

**Comment by drybjed** [comment]
After `debops` script generates the `ansible.cfg` configuration file, `ansible` command can be used and has the same functionality that `debops-task` has. I am all in favor of fixing this bug and removing `debops-task`.

**Comment by dvigueras** [comment]
Great, I'll use ansible command then.

**Comment by dvigueras** [comment]
`debops` command doesn't work well too, it looks like it's trying to read the old inventory file, instead of the file set in `ansible.cfg`.

I think it could be something related to the function `find_inventorypath`, but I'm not an expert in Python.

**Comment by tallandtree** [comment]
I'm running debops (with ansible-playbook 2.1.0.0, from fedora 21) and it looks like the myproject/.debops.cfg is not being read. If I set file-paths in .debops.cfg,

```
[paths]
file-paths: /home/myname/cm-sys-servers/ansible/files

```

`[ "{{ lookup('file', 'sshkeys/mykey.pub') }}" ]` still does not find the file, but searches in the default location only:

`fatal: [mymachine]: FAILED! => {"failed": true, "msg": "the file_name '/home/myname/.local/share/debops/debops-playbooks/playbooks/sshkeys/mykey.pub' does not exist, or is not readable"}`

**Comment by drybjed** [comment]
You need to use `lookup('file_src', 'sshkey/mykey.pub')` lookup plugin. `file` is the "normal" one.

**Comment by tallandtree** [comment]
Indeed, when I use file_src, the file is found. But there is a difference between lookup('file_src', 'sshkey/mykey.pub') and lookup('file', '/home/myname/cm-sys-servers/ansible/files/sshkeys/mykey.pub'). The latter reads contents of the file, while the first only reads the path with filename. I get an invalid key specified:

`sshkeys: [ "{{ lookup('file_src', 'sshkeys/mykey.pub') }}" ]`
gives the following error in the log:
```"sshkeys": ["/home/myname/cm-sys-servers/ansible/files/sshkeys/mykey.pub"], "state": "present", "system": false, "uid": 1001, "update_password": "on_create"}, "msg": "invalid key specified: /home/myname/cm-sys-servers/ansible/files/sshkeys/mykey.pub"}

whereas 
`sshkeys: [ "{{ lookup('file', '/home/myname/cm-sys-servers/ansible/roles/users/files/mykey.pub') }}" ]`
gives something like:

```
"sshkeys": ["ssh-rsa XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"], "state": "present", "system": false, "uid": 1001, "update_password": "on_create"}, "name": "users", "state": "present", "system": false}
```

Not sure if this is the intended behaviour or I'm doing something else wrong again. (apologies if I should be creating a new issue instead)

**Comment by drybjed** [comment]
No problem. I know what is happening here. The `file` module is used to get the contents of a specified file. The DebOps `file_src` is used to select a specific file from a set of possible paths. It was written for cases like this:

```
- copy:
    src: '{{ lookup("file_src", "relative/path/to/file") }}'
    dest: '/remote/file'
```

The main reason being, so that role contents could stay static and be able to be updated via git, and you could still modify what the role does externally via the configuration file.

In your case, you are using the `file_src` lookup in the wrong place. You would instead need to do something like this:

```
sshkeys:
  - '{{ lookup("file", lookup("file_src", "sshkeys/mykey.pub")) }}'
```

So basically, first find the absolute path to a file you want, and then get its contents. That should make it work.
